etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/PostgresDao.java

```diff
@@ -157,6 +175,18 @@ private PartnerMetadata partnerMetadataFromResultSet(ResultSet resultSet) throws
                 resultSet.getString("hash_of_message"),
                 PartnerMetadataStatus.valueOf(resultSet.getString("delivery_status")),
                 resultSet.getString("failure_reason"),
-                PartnerMetadataMessageType.valueOf(resultSet.getString("message_type")));
+                PartnerMetadataMessageType.valueOf(resultSet.getString("message_type")),
+                formatter.convertJsonToObject(
```

Comment:
I'm "OK" with the JSON formatter being here since these methods haven't been refactored to be generic yet.

Reply:
I assume we'll talk about refactoring this as well in the engineering block from the comments about `SendMessageHelper`

Reply:
Yes.  I also have this work in an unfinished state in the `story-621-read_metadata_generic` branch.